CONTRIBUTING.md

```diff
@@ -0,0 +1,106 @@
+- Does it have adequate testing?
+- Do all existing tests run unmodified?
+
+In general answers to the above should be **Yes**.  A **No** to any of them is not disqualifying of the PR, however a no answer will need an explanation and a discussion.
+ 
+There are additional considerations
+- Is the risk of accepting this change High or even Medium
+  For an acceptable PR with a high risk, we will definitely need to discuss mitigations for the risk.  A decision to upgrade the SDK or VS needs to be always low risk for our customers, they have businesses to run, they don't want to have to deal with our - risky behavior.  We may defer or delay risky PRs into a later release or abandon it.
+- Is the change as small as possible
+- Should it be chopped up into smaller chunks
+- Is the cost of reviewing the change worth the improvement made by the change
+Again, some PR’s are too big or provide too little value to merge.
+
+
+### Resources to help you get started
+
+Here are some resources to help you get started on how to contribute code or new content.
+
+- [Developers Guide](https://github.com/dotnet/fsharp/blob/main/DEVGUIDE.md) to get started on building the source code on your own.
+- [Test Guide](https://github.com/dotnet/fsharp/blob/main/TESTGUIDE.md) how to build run and work with test cases.
+- [F# compiler guide](https://github.com/dotnet/fsharp/blob/main/docs/index.md)
+- [F# language specification](https://fsharp.org/specs/language-spec/)
+- [F# language design](https://github.com/fsharp/fslang-design/)
+- [F# language suggestions](https://github.com/fsharp/fslang-suggestions/)
+- [help wanted](https://github.com/dotnet/fsharp/labels/help%20wanted) where to start
+
+### Submitting a pull request
+
+You will need to sign a [Contributor License Agreement](https://cla.dotnetfoundation.org/) when submitting your pull request. To complete the Contributor License Agreement (CLA), you will need to follow the instructions provided by the CLA bot when you send the pull request. This needs to only be done once for any .NET Foundation OSS project.
+
+If you don't know what a pull request is read this article: <https://help.github.com/articles/using-pull-requests>. Make sure the repository can build and all tests pass. Familiarize yourself with the project workflow and our coding conventions. 
+
+- **DO** ensure submissions pass all Azure DevOps legs and are merge conflict free.
+- **DO** submit language feature requests as issues in the [F# language](https://github.com/fsharp/fslang-suggestions) repos.  Please note: approved in principle does not guarantee acceptance.
+- **DO NOT** submit language features as PRs to this repo first, or they will likely be declined.
+- **DO** submit issues for other features. This facilitates discussion of a feature separately from its implementation, and increases the acceptance rates for pull requests.
+- **DO NOT** submit large code formatting changes without discussing with the team first.
+
```

Comment:
I think we could also mention that the release notes should be updated following https://keepachangelog.com/en/1.1.0/

Reply:
There's a bot that checks release notes.  And I believe the PR template has it too.

    > Release notes files:
    > - If anything under `src/Compiler` has been changed, please make sure to make an entry in `docs/release-notes/.FSharp.Compiler.Service/<version>.md`, where `<version>` is usually "highest" one, e.g. `42.8.200`
    > - If language feature was added (i.e. `LanguageFeatures.fsi` was changed), please add it to `docs/release-notes/.Language/preview.md`
    > - If a change to `FSharp.Core` was made, please make sure to edit `docs/release-notes/.FSharp.Core/<version>.md` where version is "highest" one, e.g. `8.0.200`.

    > Information about the release notes entries format can be found in the [documentation](https://fsharp.github.io/fsharp-compiler-docs/release-notes/About.html).
    > Example:
    > * More inlines for Result module. ([PR #16106](https://github.com/dotnet/fsharp/pull/16106))
    > * Correctly handle assembly imports with public key token of 0 length. ([Issue #16359](https://github.com/dotnet/fsharp/issues/16359), [PR #16363](https://github.com/dotnet/fsharp/pull/16363))
    > *`while!` ([Language suggestion #1038](https://github.com/fsharp/fslang-suggestions/issues/1038), [PR #14238](https://github.com/dotnet/fsharp/pull/14238))

    > **If you believe that release notes are not necessary for this PR, please add `NO_RELEASE_NOTES` label to the pull request.**